connector/kinesis-asl-assembly/pom.xml

```diff
@@ -62,12 +62,18 @@
     <dependency>
       <groupId>com.google.protobuf</groupId>
       <artifactId>protobuf-java</artifactId>
-      <version>2.6.1</version>
-      <!-- 
-         We are being explicit about version here and overriding the 
-         spark default of 2.5.0 because KCL appears to have introduced 
-         a dependency on protobuf 2.6.1 somewhere between KCL 1.4.0 and 1.6.1.
-       -->
+      <scope>compile</scope>
+    </dependency>
+      <!--
+         Set the guava version explicitly and override the spark default
+         because KCL 2.x is not compatible with guava 14.0.1 as specified
+          in <guava.version>.
+      -->
+    <dependency>
+      <groupId>com.google.guava</groupId>
+      <artifactId>guava</artifactId>
+      <version>${aws.kinesis.client.guava.version}</version>
+      <scope>compile</scope>
```

Comment:
Ur, I'm not sure this is okay or not, @junyuc25 .

Reply:
cc @dongjoon-hyun

Reply:
Could you share more about your concern here @dongjoon-hyun ? Thanks!

Reply:
Sorry for being late.

1. Is this only for Kinesis or for all AWS SKD v2?
2. Instead of the following, can we use the latest33.0.0-jre like #44795 ?
```
<aws.kinesis.client.guava.version>32.1.1-jre</aws.kinesis.client.guava.version>
```

Reply:
1. This change is only for the Kinesis connector modules. 
2. Yes I changed it to use the same Guava version as the one used in other module.